app/controllers/project/Versions.scala

```diff
@@ -141,31 +156,37 @@ class Versions @Inject()(stats: StatTracker,
     * @return View of project
     */
   def showList(author: String, slug: String, channels: Option[String], page: Option[Int]) = {
-    ProjectAction(author, slug) { implicit request =>
+    ProjectAction(author, slug).async { request =>
       val project = request.project
-      val allChannels = project.channels.toSeq
-
-      var visibleNames: Option[Array[String]] = channels.map(_.toLowerCase.split(','))
-      val visible: Option[Array[Channel]] = visibleNames.map(_.map { name =>
-        allChannels.find(_.name.equalsIgnoreCase(name))
-      }).map(_.flatten)
-
-      val visibleIds: Array[Int] = visible.map(_.map(_.id.get)).getOrElse(allChannels.map(_.id.get).toArray)
-
-      val pageSize = this.config.projects.get[Int]("init-version-load")
-      val p = page.getOrElse(1)
-      val versions = project.versions.sorted(
-        ordering = _.createdAt.desc,
-        filter = _.channelId inSetBind visibleIds,
-        offset = pageSize * (p - 1),
-        limit = pageSize)
-
-      if (visibleNames.isDefined && visibleNames.get.toSet.equals(allChannels.map(_.name.toLowerCase).toSet)) {
-        visibleNames = None
-      }
+      implicit val r = request.request
+      project.p.channels.toSeq.flatMap { allChannels =>
+        var visibleNames: Option[Array[String]] = channels.map(_.toLowerCase.split(','))
+        val visible: Option[Array[Channel]] = visibleNames.map(_.map { name =>
+          allChannels.find(_.name.equalsIgnoreCase(name))
+        }).map(_.flatten)
+
+        val visibleIds: Array[Int] = visible.map(_.map(_.id.get)).getOrElse(allChannels.map(_.id.get).toArray)
+
+        val pageSize = this.config.projects.get[Int]("init-version-load")
+        val p = page.getOrElse(1)
+        val futureVersions = project.p.versions.sorted(
+          ordering = _.createdAt.desc,
+          filter = _.channelId inSetBind visibleIds,
+          offset = pageSize * (p - 1),
+          limit = pageSize)
+
+        if (visibleNames.isDefined && visibleNames.get.toSet.equals(allChannels.map(_.name.toLowerCase).toSet)) {
+          visibleNames = None
+        }
 
-      this.stats.projectViewed { implicit request =>
-        Ok(views.list(project, allChannels, versions, visibleNames, p))
+        for {
+          versions <- futureVersions
+          r <- this.stats.projectViewed { request =>
+            Ok(views.list(project, allChannels, versions, visibleNames, p))
+          }(request)
```

Comment:
Any reason request is no longer marked as implicit, but instead passed in explicitly here?